src/api/helpers/index.ts

```diff
@@ -1,29 +1,37 @@
-import { DEVELOPMENT, EXTENSION_ID } from "statics";
+import {
+  DEVELOPMENT,
+  EXTERNAL_MSG_REQUEST,
+  EXTERNAL_MSG_RESPONSE,
+} from "statics";
+import { Response } from "../types";
 
-interface Response {
-  applicationState: string;
-  publicKey: string;
-  mnemonicPhrase: string;
-  isCorrectPhrase: boolean;
-  confirmedPassword: boolean;
-  transactionStatus: string;
-  error: string;
-}
-
-export const sendMessageAndAwaitResponseExternal = (msg: {}): Promise<Response> =>
-  new Promise((resolve) => {
-    chrome.runtime.sendMessage(EXTENSION_ID, msg, (res: Response) =>
-      resolve(res),
-    );
+export const sendMessageToContentScript = (msg: {}): Promise<Response> => {
+  window.postMessage(
+    { source: EXTERNAL_MSG_REQUEST, ...msg },
+    window.location.origin,
+  );
+  return new Promise((resolve) => {
+    const messageListener = (event: { source: any; data: Response }) => {
+      // We only accept messages from ourselves
+      if (event.source !== window) return;
+      // Only respond to messages tagged as being from our content script
+      if (!event.data.source || event.data.source !== EXTERNAL_MSG_RESPONSE) {
+        return;
+      }
+      resolve(event.data);
+      window.removeEventListener("message", messageListener);
```

Comment:
Content Script takes Background's response, posts a message to itself. Since it shares the same window as Lyra API, it is able to listen (similar to above) and then resolve the API request with the data from Content Script